pkg/route/ingress/ingress.go

```diff
@@ -444,7 +468,8 @@ func (c *Controller) sync(key queueKey) error {
 				continue
 			}
 
-			if routeMatchesIngress(existing, ingress, &rule, &path, c.secretLister, c.serviceLister, host, hostIsWildcard) {
+			_, match := routeMatchesIngress(existing, ingress, &rule, &path, c.secretLister, c.serviceLister, host, hostIsWildcard, ingressToRouteRule)
```

Comment:
You should assign back to `ingressToRouteRule`. Or you can let `routeMatchesIngress` return its incomplete rules and assign them to `ingressToRouteRule` as follows:
```suggestion
			match, incompleteMatchRules := routeMatchesIngress(existing, ingress, &rule, &path, c.secretLister, c.serviceLister, host, hostIsWildcard)
			ingressToRouteRule = append(ingressToRouteRule, incompleteMatchRules...)
```